driver/src/contracts/stablex_contract.rs

```diff
@@ -162,17 +162,63 @@ impl StableXContract for StableXContractImpl {
                 .ok_or_else(|| anyhow!("block {:?} is missing", block_number))
         };
 
-        let mut current_block = get_block(BlockNumber::Latest)?;
-        let mut block_number = current_block
+        struct Bounds {
+            lower: u64,
+            upper: u64,
+        }
+        impl Bounds {
+            fn diff(&self) -> u64 {
+                self.upper - self.lower
+            }
+            fn mid(&self) -> u64 {
+                self.diff() / 2
+            }
+        }
+
+        let get_bounds_for_binary_search =
```

Comment:
Maybe this is personal preference, but I would call this `binary_search_bounds` (or anything without `get_` for that matter.